static/src/javascripts/projects/journalism/services/viewport.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import {
+    observe as observeIO,
+    unobserve as unobserveIO,
+} from './impl/viewport-io';
+import {
+    observe as observeScroll,
+    unobserve as unobserveScroll,
+} from './impl/viewport-scroll';
+
+const viewport: ViewportService =
+    'IntersectionObserver' in window
```

Comment:
intersection observer is available in polyfill.io?

Reply:
I ask you 😅

Reply:
[it is](https://polyfill.io/v2/docs/features/#IntersectionObserver). 

if atoms need things that are missing in frontend, it might be a better policy to look at adding them to frontend first, and then use them from there. 

otherwise it will be very easy to arrive a situation where we're shipping multiple instances of the same functionality to clients which will only add to our attack on our users CPUs.

also, given we have a polyfill service exactly to avoid shipping polyfill code where it's not needed (in this case, over half of our users), that should always be our first port of call

Reply:
100% agree, this code is mostly taken out of the work I did for the apps. I don't need to feature detect then, mucho gusto 🤗